tracks/erlang/exercises/bob/mentoring.md

```diff
@@ -0,0 +1,46 @@
+### Reasonable solutions
+
+```erl
+response(Str) ->
+  response(Str, false, false, false, false).
+
+response([], false, false, false, false) ->
+  "Fine. Be that way!";
+response([], true, false, _HasOther, false) ->
+  "Whoa, chill out!";
+response([], true, false, _HasOther, true) ->
+  "Calm down, I know what I'm doing!";
+response([], _HasUpper, _HasLower, _HasOther, false) ->
+  "Whatever.";
+response([], _HasUpper, _HasLower, _HasOther, true) ->
+  "Sure.";
+response([C|Str], HasUpper, HasLower, HasOther, MaybeQuestion) when C=<$\s ->
+  response(Str, HasUpper, HasLower, HasOther, MaybeQuestion);
+response([$?|Str], HasUpper, HasLower, HasOther, _MaybeQuestion) ->
+  response(Str, HasUpper, HasLower, HasOther, true);
+response([C|Str], _HasUpper, HasLower, HasOther, _MaybeQuestion) when C>=$A, C=<$Z ->
+  response(Str, true, HasLower, HasOther, false);
+response([C|Str], HasUpper, _HasLower, HasOther, _MaybeQuestion) when C>=$a, C=<$z ->
+  response(Str, HasUpper, true, HasOther, false);
+response([_|Str], HasUpper, HasLower, _HasOther, _MaybeQuestion) ->
+  response(Str, HasUpper, HasLower, true, false).
+```
```

Comment:
You say that there are many reasonable solutions. If this is the average size of a reasonable solution, I wouldn't mind if there were 1-3 of them. But I understand your incentive.

As for this particular solution, I believe that you can make it more abstract and thus more readable if you apply meaningful names in either helper functions or variables. The code doesn't read well from top to bottom with the first many default cases being nonsensical until you reach the bottom.

I see that the [example solution](https://github.com/exercism/erlang/blob/master/exercises/bob/src/example.erl) does this.

Perhaps, if you'd like a single-pass solution that does no trimming, it could export the particular properties that `response` needs; something like...

```erlang
response(Str) ->
  {IsSilent, IsShouting, IsQuestion} = classify(Str),
  if IsSilent -> "Fine. Be that way!";
     IsShouting andalso IsQuestion -> ...;
     ...
  end.
```

or an equivalent case-of.

Reply:
Or... one that used atoms rather than booleans for the classification, I suppose.